test/models/patient_duplicate_detection_test.rb

```diff
@@ -0,0 +1,187 @@
+# frozen_string_literal: true
+
+require 'test_case'
+
+class PatientDuplicateDetectionTest < ActiveSupport::TestCase
+  def setup
+    Patient.destroy_all
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: '1996-12-31', sex: 'Male', user_defined_id_statelocal: 'asdf123')
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: '1996-12-31', sex: nil)
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: nil, sex: 'Male')
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: nil, sex: nil)
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: nil, sex: 'Male')
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: '1996-12-31', sex: nil)
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: '1996-12-31', sex: nil)
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: nil, sex: nil)
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: nil, sex: nil)
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: nil, sex: nil)
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: '1996-12-30', sex: nil)
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: '1996-12-31', sex: 'Female')
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: '1996-12-30', sex: 'Male')
+    create(:patient, first_name: 'Deckard', last_name: 'Cain', date_of_birth: nil, sex: 'Female')
+    create(:patient, first_name: 'Deckar', last_name: 'Cain', date_of_birth: '1996-12-31', sex: 'Male')
+    create(:patient, first_name: 'Deckard', last_name: 'Cai', date_of_birth: '1996-12-31', sex: 'Male')
+  end
+
+  test 'duplicate_data finds duplicate with input of FN LN Sex DoB ID' do
+    patient_dup = Patient.new(first_name: 'Deckard', last_name: 'Cain', date_of_birth: '1996-12-31', sex: 'Male', user_defined_id_statelocal: 'asdf123')
```

Comment:
I think we can actually not have to deal with calling `&.strftime('%F')` on `date_of_birth` later by defining `patient_dup` as a ruby object, sorry for telling you to create it as a new `Patient` earlier, it seems like it's just a ruby object (where `date_of_birth` is not yet casted from a string to a date) instead of a `Patient` record that is passed to the `duplicate_date` method. So we can write:

```
patient_dup = { first_name: 'Deckard', last_name: 'Cain', date_of_birth: '1996-12-31', sex: 'Male', user_defined_id_statelocal: 'asdf123' }
```

and then remove `&.strftime('%F')` from line 32